src/v/security/oidc_service.cc

```diff
@@ -0,0 +1,226 @@
+/*
+ * Copyright 2023 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+#include "security/oidc_service.h"
+
+#include "http/client.h"
+#include "security/jwt.h"
+#include "security/logger.h"
+#include "ssx/future-util.h"
+
+#include <seastar/core/lowres_clock.hh>
+#include <seastar/core/sstring.hh>
+
+#include <absl/algorithm/container.h>
+#include <boost/url/parse.hpp>
+#include <boost/url/url.hpp>
+
+namespace security::oidc {
+
+namespace {
+ss::future<ss::sstring> make_request(std::string_view url_view) {
+    boost::urls::url url{url_view};
```

Comment:
> question: Just checking that we're good with allowing exceptions to leak from here? I think it will be caught through the configuration change?

I'll check.

Reply:
Exceptions are caught and logged